tikv-client/src/main/java/com/pingcap/tikv/region/RegionStoreClient.java

```diff
@@ -190,25 +250,54 @@ private ByteString getHelper(GetResponse resp) {
             .setStartKey(startKey)
             .setVersion(version)
             .setKeyOnly(keyOnly)
-            .setLimit(getConf().getScanBatchSize())
+            .setLimit(session.getConf().getScanBatchSize())
             .build();
 
     KVErrorHandler<ScanResponse> handler =
         new KVErrorHandler<>(
             regionManager, this, region, resp -> resp.hasRegionError() ? resp.getRegionError() : null);
     ScanResponse resp = callWithRetry(backOffer, TikvGrpc.METHOD_KV_SCAN, request, handler);
-    return scanHelper(resp);
+    return scanHelper(resp, backOffer);
   }
 
-  private List<KvPair> scanHelper(ScanResponse resp) {
+  // TODO: remove helper and change to while style
+  // needs to be fixed as batchGet
+  // which we shoule retry not throw
+  // exception
+  private List<KvPair> scanHelper(ScanResponse resp, BackOffer bo) {
```

Comment:
You are using the same logic as `batchGetHelper`. Try to merge them to eliminate redundant code.

Reply:
I think this may be resolved after you have finished your logic because the Helper needs to be deprecated. And merge them and replace with while loop style.
We can present an issue for it